routers/user/auth.go

```diff
@@ -120,6 +127,11 @@ func SignInPost(ctx *context.Context, form auth.SignInForm) {
 	// Clear whatever CSRF has right now, force to generate a new one
 	ctx.SetCookie(setting.CSRFCookieName, "", -1, setting.AppSubUrl)
 
+	if redirectTo := ctx.Query("redirect_to"); len(redirectTo) > 0 {
```

Comment:
I don't understand either. All I know is it takes this code to make
the redirect work. There are too many layers I don't fully understand
to take care of cleanups / refactoring.